### PR TITLE
[CPDLP-2548] Add search AB users feature to admin dashboard

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

- Ticket: [CPDLP-2548](https://dfedigital.atlassian.net/browse/CPDLP-2548)

### Changes proposed in this pull request

Add search AB users feature to admin dashboard.

[CPDLP-2548]: https://dfedigital.atlassian.net/browse/CPDLP-2548?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/admin/appropriate_bodies/users_controller.rb [hunk 1]
```diff
@@ -67,6 +67,14 @@ def set_appropriate_body_profile_and_authorize
         @appropriate_body_profile = AppropriateBodyProfile.find(params[:id])
         authorize @appropriate_body_profile
       end
+
+      def scoped
+        return policy_scope(AppropriateBodyProfile).includes(:user).order("users.full_name asc") if params[:query].blank?
```
Comment (ethax-ross) [inline app/controllers/admin/appropriate_bodies/users_controller.rb:72]: Worth encapsulating this check in the `SearchQuery` so it can be tested?

File: spec/features/admin/appropriate_bodies/users_spec.rb [hunk 2]
```diff
@@ -109,11 +134,19 @@ def and_i_see(string)
...
+
   def and_i_should_not_see(string)
     expect(page).not_to have_content(string)
   end
 
   def given_a_appropriate_body_user_exists
     @appropriate_body_user = create(:user, :appropriate_body)
   end
+
+  def when_i_fill_in(selector, with:)
```
Comment (ethax-ross) [inline spec/features/admin/appropriate_bodies/users_spec.rb:149]: One for another PR but we seem to duplicate a lot of these helper methods; we could DRY it all up if we extract them

File: spec/queries/appropriate_body_profiles/search_query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    context "when the query includes part of the email of a user" do
+      let(:query) { "user_a@" }
+
+      it "searches users by email" do
+        expect(subject).to match_array([appropriate_body_profile_a])
+      end
+    end
+
+    context "when the query includes part of the appropriate boby name of a user" do
```
Comment (ethax-ross) [inline spec/queries/appropriate_body_profiles/search_query_spec.rb:33]: `appropriate boby` 😅

File: spec/queries/appropriate_body_profiles/search_query_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    let(:user_b) { create(:user, email: "user_c@nothing.com", full_name: "User B") }
+    let(:user_c) { create(:user, email: "user_b@nothing.com", full_name: "User C") }
+    let(:appropriate_body) { create(:appropriate_body_local_authority, name: "Test") }
+    let!(:appropriate_body_profile_a) { create(:appropriate_body_profile, appropriate_body:, user: user_a) }
+    let!(:appropriate_body_profile_b) { create(:appropriate_body_profile, appropriate_body:, user: user_b) }
+    let!(:appropriate_body_profile_c) { create(:appropriate_body_profile, appropriate_body:, user: user_c) }
+
+    subject { described_class.new(query:).call }
+
+    context "when the query includes part of the name of a user" do
```
Comment (ethax-ross) [inline spec/queries/appropriate_body_profiles/search_query_spec.rb:17]: Worth adding a test for when there are no matches to ensure it returns empty?

File: spec/queries/appropriate_body_profiles/search_query_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      it "searches users by email" do
+        expect(subject).to match_array([appropriate_body_profile_a])
+      end
+    end
+
+    context "when the query includes part of the appropriate boby name of a user" do
+      let(:query) { "Test" }
+
+      it "searches users by appropriate body name" do
+        expect(subject).to match_array([appropriate_body_profile_a, appropriate_body_profile_b, appropriate_body_profile_c])
```
Comment (ethax-ross) [inline spec/queries/appropriate_body_profiles/search_query_spec.rb:37]: Would this be better as an `eq` to cover the ordering?

**Review by ethax-ross — COMMENTED** [review]
Looks great just some minor suggestions 👍

File: spec/queries/delivery_partner_profiles/search_query_spec.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe DeliveryPartnerProfiles::SearchQuery do
```
Comment (cwrw) [inline spec/queries/delivery_partner_profiles/search_query_spec.rb:5]: +1 for adding missing tests for this!

**Review by cwrw — COMMENTED** [review]
nice, past the other comments works great

**Review by cwrw — APPROVED** [review]
